src/App.tsx

```diff
@@ -1,18 +1,226 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useContext, useEffect, useState,
+} from 'react';
+
+import classNames from 'classnames';
+
+import {
+  CustomTodo,
+  getTodos,
+  addTodo,
+  deleteTodo,
+  updateTodo,
+} from './api/todos';
+import { Filter } from './types/Filter';
+
 import { AuthContext } from './components/Auth/AuthContext';
+import { Footer } from './components/Footer/Footer';
+import Form from './components/Form/Form';
+import TodoList from './components/TodoList/TodoList';
+import Warning from './components/Warning/Warning';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const [todosFromTheServer, setTodosFromTheServer] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [errorCount, setErrorCount] = useState(0);
+  const [currentFilter, setCurrentFilter] = useState<Filter>(Filter.ALL);
+  const [isAdding, setIsAdding] = useState(false);
+  const [isToggling, setIsToggling] = useState(false);
+  const [isChanging, setIsChanging] = useState(false);
+  const [deletingCompleted, setDelitingCompleted] = useState(false);
+  const [customTodo, setCustomTodo] = useState<CustomTodo | null>(null);
+
   const user = useContext(AuthContext);
-  const newTodoField = useRef<HTMLInputElement>(null);
+
+  const areAllCompleted = todosFromTheServer.filter(
+    todoFromServer => !todoFromServer.completed,
+  ).length === 0;
+
+  const onFilter = (type: Filter) => setCurrentFilter(type);
+
+  const filterTodos = () => {
+    switch (currentFilter) {
+      case Filter.COMPLETED: {
+        const completedTodos = todosFromTheServer.filter(
+          todo => todo.completed,
+        );
+
+        return completedTodos;
+        break;
+      }
+
+      case Filter.ACTIVE: {
+        const activeTodos = todosFromTheServer.filter(
+          todo => !todo.completed,
+        );
+
+        return activeTodos;
+        break;
+      }
+
+      default: {
+        return todosFromTheServer;
+      }
+    }
+  };
+
+  let todosToShow = filterTodos();
```

Comment:
it's bad practice to use variable, and change it for rendering some content. Don't do that. 
you can use `useState` and others hooks